src/Symfony/Component/Lock/Store/ZookeeperStore.php

```diff
@@ -0,0 +1,239 @@
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function putOffExpiration(Key $key, $ttl)
+    {
+    }
+
+    /**
+     * @param int    $errNo  Error number
+     * @param string $errStr Error String
+     *
+     * @throws LockConflictedException
+     * @throws LockAcquiringException
+     */
+    public static function errorHandlerForCreateNode($errNo, $errStr)
+    {
+        if (E_WARNING === $errNo && 'Zookeeper::create(): error: node exists' === $errStr) {
+            throw new LockConflictedException($errStr);
+        }
+
+        throw new LockAcquiringException($errStr);
+    }
+
+    /**
+     * Create a zookeeper node.
+     * If it has a path that doesn't exists, then this method creates the path with the same permissions before
+     * creating a node.
+     *
+     * @param string $node  The node which needs to be created
+     * @param string $value The value to be assigned to a zookeeper node
+     * @param array  $acl   Permissions settings on a zookeeper node
+     * @param int    $type  Type of zookeeper node OPTIONAL
+     *
+     * @return string Path of the node created
+     *
+     * @throws LockConflictedException
+     * @throws LockAcquiringException
+     */
+    private function createNewLock(
+        string $node,
+        string $value = '',
+        array $acl = array(),
+        int $type = Zookeeper::EPHEMERAL
+    ): string {
+        set_error_handler(array(self::class, 'errorHandlerForCreateNode'));
+        try {
+            // If the node exists then update the value of the node.
+            if ($this->zookeeper->exists($node)) {
+                throw new LockConflictedException('Lock already exists');
+            }
+
+            // Default Node Permissions
+            if (empty($acl)) {
+                $acl = array(array('perms' => Zookeeper::PERM_ALL, 'scheme' => 'world', 'id' => 'anyone'));
+            }
+
+            // Check if there is a path that needs to be created before creating a node.
+            $parts = array_filter(explode('/', $node));
+            $subpath = '';
+
+            // If a node needs to be created with the path "/node1/node2/node3",
+            // this loop will create "/node1" and "/node1/node2" before creation of deepest level node.
+            for ($numberOfParts = count($parts); $numberOfParts > 1; $numberOfParts = count($parts)) {
+                $subpath .= '/'.array_shift($parts);
+
+                if ($this->zookeeper->exists($subpath)) {
+                    continue;
+                }
+
+                $nodeCreated = $this->zookeeper->create($subpath, null, $acl);
+                if (empty($nodeCreated)) {
+                    throw new LockAcquiringException('Unable to create node for '.$subpath);
+                }
+            }
+
+            $nodeCreated = $this->zookeeper->create($node, $value, $acl, $type);
+            if (!$nodeCreated) {
+                throw new LockAcquiringException('Unable to create node for '.$subpath);
```

Comment:
SHouldn't it be a `LockConflictException`?

Reply:
If you see the error handler, if another node was created due to a race condition, it would throw a `LockConflictedException`. If it reaches here, it means it failed to acquire the lock for some other reason.